src/schnetpack/nn/embed.py

```diff
@@ -0,0 +1,278 @@
+                geometric_data.update(_init_sphc_zeros(z=z,
+                                                       sph_ij=sph_harms_ij,
+                                                       phi_r_cut=phi_r_cut,
+                                                       idx_i=idx_i,
+                                                       point_mask=point_mask,
+                                                       mp_normalization=self._lambda)
+                                      )
+            else:
+                # Initialize SPHCs with a neighborhood dependent embedding
+                geometric_data.update(_init_sphc(z=z,
+                                                 sph_ij=sph_harms_ij,
+                                                 phi_r_cut=phi_r_cut,
+                                                 idx_i=idx_i,
+                                                 point_mask=point_mask,
+                                                 mp_normalization=self._lambda)
+                                      )
+
+        # Solid harmonics (Spherical harmonics + radial part)
+        if self.solid_harmonic:
+            rbf_ij = safe_scale(rbf_ij, scale=phi_r_cut[:, None])  # shape: (n_pairs,K)
+            g_ij = sph_harms_ij[:, :, None] * rbf_ij[:, None, :]  # shape: (n_pair,m_tot,K)
+            g_ij = safe_scale(g_ij, scale=pair_mask[:, None, None], placeholder=0)  # shape: (n_pair,m_tot,K)
+            geometric_data.update({'g_ij': g_ij})
+
+        return geometric_data
+
+
+class AtomTypeEmbed(nn.Module):
+    def __init__(self, features: int, prop_keys: Dict, num_embeddings: int = 100,
+                 module_name: str = 'atom_type_embed', *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.features = features
+        self.prop_keys = prop_keys
+        self.module_name = module_name
+        self.num_embeddings = num_embeddings
+        self.atomic_type_key = self.prop_keys.get('atomic_type')
+
+    def forward(self, inputs: Dict, *args, **kwargs) -> np.ndarray:
+        """
+        Create atomic embeddings based on the atomic types.
+
+        Args:
+            inputs (Dict):
+                z (Array): atomic types, shape: (n)
+                point_mask (Array): Mask for atom-wise operations, shape: (n)
+            *args (Tuple):
+            **kwargs (Dict):
+
+        Returns: Atomic embeddings, shape: (n,F)
+
+        """
+        z = inputs[self.atomic_type_key]
+        point_mask = inputs[properties.point_mask]
+        z = z.astype(np.int32)  # shape: (n)
+        return safe_scale(nn.Embedding(num_embeddings=self.num_embeddings, embedding_dim=self.features)(z),
+                          scale=point_mask[:, None])
+
+    def __dict_repr__(self):
+        return {self.module_name: {'num_embeddings': self.num_embeddings,
+                                   'features': self.features,
+                                   'prop_keys': self.prop_keys}}
+
+
+class OneHotEmbed(nn.Module):
```

Comment:
We can keep this.
Have you tested if this implementation and the original implementation result in the same initial one hot embedding ?
Please make sure to always use torch instead of numpy